operators/test/e2e/test/builder.go

```diff
@@ -13,6 +13,8 @@ type Builder interface {
 	CheckK8sTestSteps(k *K8sClient) StepList
 	// CheckStackTestSteps returns all test steps to verify the given resource is running as expected
 	CheckStackTestSteps(k *K8sClient) StepList
+	// UpgradeTestSteps returns all the steps necessary to upgrade an existing resource to new configuration.
+	UpgradeTestSteps(k *K8sClient) StepList
```

Comment:
The distinction with `MutationTestSteps` is not obvious. Also `configuration` is a bit misleading, sounds like ES configuration.
How about:

* SafeMutationTestSteps (corresponding to the current MutationTestSteps)
* MutationReversalTestSteps (corresponding to what you added in this PR)

The single `UpgradeTestSteps ` that simply applies a resource sounds a bit overkill to be part of the `Builder` interface. I think it would be fine to extract the function as util somewhere, to be used in both `SafeMutationTestSteps` and `MutationReversalTestSteps`.

Reply:
I don't like the fact that we are tying the test steps to run to the builder. I expect the builder to contain crud bits for the different resource types we want to test. But implementing the tests itself, or at least the basic structure of the tests (for mutation and as you suggest for these back-and-forth style reversal tests) seems overly limiting. 

I would expect us to need different kinds of scenarios or variants of scenarios for these tests, implementing them on the builder allows for one mutation scenario (which can be parameterized ofc) but it feels like a very rigid structure.

I guess what I would expect is that we refactor `MutationTestSteps` and remove it from the builder as well, it has nothing to do with building test resources but actually encodes a test recipe.

Reply:
Agreed 👍